### DISCUSSION TITLE
Inheritance on Notifier with code generation

### DISCUSSION THREAD

**Discussion by logic-and-math** [rrousselGit/riverpod#3091] [General]
Starting a new project and trying v2 for the first time.
Previously a lot of my code relied on state notifier inheritance.

```dart
abstract class PaginatedNotifier<T> extends StateNotifier<PaginatedState<T>> {
//code of the abstract class
}

class CarsNotifier extends PaginatedNotifier<Car> {
//code ...
} 

```

This allowed me to keep the common logic related to pagination in a single abstract notifier.

Now I have this notifier:
```dart
@Riverpod()
class CarNotifier extends _$CarNotifier {
  @override
  PaginatedState build(String config) {
    return PaginatedState();
  }
}
```

How can I do the same in v2 with code generation? Thank you

**Comment by benji-farquhar** [comment]
Anyone done this with Notifier without code gen?
